data_steward/cdr_cleaner/cleaning_rules/null_invalid_foreign_keys.py

```diff
@@ -39,11 +39,11 @@
     ON t.{{field}} = {{prefix}}.{{field}}""")
 
 SANDBOX_QUERY = JINJA_ENV.from_string("""
-    CREATE OR REPLACE TABLE
+    CREATE OR REPLACE TABLE 
     `{{project_id}}.{{sandbox_dataset_id}}.{{intermediary_table}}` AS (
-    SELECT {{cols}}
-    FROM `{{project_id}}.{{dataset_id}}.{{table_name}}` t
-    {{join_exp}})""")
+        SELECT t.* FROM `{{project_id}}.{{dataset_id}}.{{table_name}}` AS t 
+        LEFT JOIN {{dataset_id}}.{{table}} AS tc USING({{cols}}) 
+        WHERE tc.{{cols}} IS NULL)""")
```

Comment:
@dcampbell-vumc this query is will not work since some tables have multiple foreign keys that need to be joined to different tables. so for each table, the join expression will be different. and {{cols}} is a list of all the columns for any given table.  So the join won't work. Rather you should be updating the join expression that is being generated for each table to filter the invalid keys.